src/vos/vos_obj.h

```diff
@@ -238,4 +241,16 @@ vos_obj_discard_hold(struct daos_lru_cache *occ, struct vos_container *cont, dao
 void
 vos_obj_discard_release(struct daos_lru_cache *occ, struct vos_object *obj);
 
+static inline bool
+vos_obj_is_empty(struct vos_object *obj)
+{
+	return !obj->obj_df || obj->obj_df->vo_tree.tr_class == 0;
+}
+
+static inline bool
+vos_obj_flattened(struct vos_object *obj)
+{
+	return obj->obj_df && obj->obj_df->vo_sync == DAOS_EPOCH_MAX;
```

Comment:
we don't have use case of epoch_max in vos_object_df today?

Reply:
now the "vo_sync" only updated by "vos_dtx_mark_sync" and the reasonable value should be [0, current HLC/epoch], the EPOCH_MAX seems impossible to be used in normal case.
@Nasf-Fan could you please confirm if it is reasonable to use EPOCH_MAX to differentiate flattened obj? thanks

Reply:
just checked offline with @Nasf-Fan , we don't have use case of EPOCH_MAX in vos_obj_df::vo_sync today.

Reply:
Yes, sync logic will not use "EPOCH_MAX".